src/shared/components/query/StudySearch.tsx

```diff
@@ -0,0 +1,114 @@
+import {
+    addClauses,
+    removePhrase,
+    toQueryString,
+} from 'shared/lib/query/textQueryUtils';
+import * as React from 'react';
+import { FunctionComponent, useState } from 'react';
+import {
+    SearchClause,
+    QueryUpdate,
+} from 'shared/components/query/filteredSearch/SearchClause';
+import { QueryParser } from 'shared/lib/query/QueryParser';
+import _ from 'lodash';
+import { SearchBox } from 'shared/components/query/filteredSearch/SearchBox';
+import { StudySearchControls } from 'shared/components/query/filteredSearch/StudySearchControls';
+import { observer, useLocalObservable } from 'mobx-react-lite';
+
+export type AutosuggestStudySearchProps = {
+    parser: QueryParser;
+    query: SearchClause[];
+    onSearch: (query: SearchClause[]) => void;
+};
+
+export const StudySearch: FunctionComponent<AutosuggestStudySearchProps> = observer(
+    function(props) {
+        const store = useLocalObservable(() => ({
+            isMenuOpen: false,
+            toggle() {
+                this.isMenuOpen = !this.isMenuOpen;
+            },
+        }));
+
+        return (
+            <div
+                data-test="study-search"
+                className={`dropdown ${store.isMenuOpen ? 'open' : ''}`}
+            >
+                <div className="input-group input-group-sm input-group-toggle">
+                    <SearchBox
+                        queryString={toQueryString(props.query)}
+                        onType={handleQueryTyping}
+                    />
+                    <SearchMenuToggle onClick={store.toggle} />
+                </div>
+                <ClearSearchButton
+                    show={props.query.length > 0}
+                    onClick={() => handleQueryTyping('')}
+                />
+                <StudySearchControls
+                    query={props.query}
+                    filterConfig={props.parser.searchFilters}
+                    onChange={handleQueryUpdate}
+                    parser={props.parser}
+                />
+            </div>
+        );
+
+        function handleQueryTyping(update: string) {
```

Comment:
interesting that you put these functions down below the return.  i think it works because of hoisting, but it's a bit confusing to me to read.  in any case, these probably should be outside of the function component or you can use the useCallback hook to make sure that these functions are not unnecessarily redeclared every time a component renders.   this is sort of the curse of funtional components.  that we no longer have "methods"

Reply:
i would be in favor of moving them outside the function and then passing them the props they need (or even props)

Reply:
Solved in d796045b2